### ISSUE TITLE
Callback arg in setString never used

### ISSUE THREAD

**Issue by eliot1019** [ammarahm-ed/react-native-mmkv-storage#215]
It seems like the callback argument from TS code is completely ignored for the synchronous setString API.

https://github.com/ammarahm-ed/react-native-mmkv-storage/blob/fb0f3224aaba77e0c3fbef1f1e4b3142bd42ebb6/index.d.ts#L197-L208

https://github.com/ammarahm-ed/react-native-mmkv-storage/blob/fb0f3224aaba77e0c3fbef1f1e4b3142bd42ebb6/src/api.js#L91-L93

**Comment by ammarahm-ed** [comment]
@eliot1019 the callback arg was used in older versions of this library. They need to be removed now as they are not used actually.  I will work on cleanup.

**Comment by eliot1019** [comment]
The callback seems useful though: for instance if a write fails and the callback is given a boolean `false`, we could do things like retry, log the failure to telemetry, or maybe even clear part of storage.

**Comment by ammarahm-ed** [comment]
@eliot1019 I agree on that. But callbacks should be truly async in that case so the program can continue. I will see what best approach can be used.

**Comment by eliot1019** [comment]
As it stands right now, it doesn't seem like there is any way for us to know if a write has succeeded or not. 
It also doesn't seem like the core MMKV lib returns error msgs on write, just success/failure which is unfortunate.

**Comment by ammarahm-ed** [comment]
@eliot1019 if write succeeds, it should return true, otherwise it should return undefined
